src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -882,6 +890,10 @@ public async Task CopyFromSourceToDestinationAsync()
 
                     // Now, use the source stream's CopyToAsync to push directly to our inflater via this helper stream
                     await _deflateStream._stream.CopyToAsync(this, _arrayPoolBuffer.Length, _cancellationToken).ConfigureAwait(false);
+                    if (!_deflateStream._inflater.Finished())
+                    {
+                        ThrowGenericInvalidData();
+                    }
```

Comment:
These conditions still hold true in CopyTo[Async].
```suggestion
                    if (!_deflateStream._inflater.Finished())
                    {
                        Debug.Assert(_deflateStream._inflater.NonEmptyInput());
                        Debug.Assert(_deflateStream._inflater.AvailableOutput > 0);
                        ThrowGenericInvalidData();
                    }
```